compiler_opt/rl/problem_configuration.py

```diff
@@ -100,6 +105,143 @@ def get_nonnormalized_features(self) -> Iterable[str]:
   def get_runner_type(self) -> 'type[compilation_runner.CompilationRunner]':
     raise NotImplementedError
 
+  @abc.abstractmethod
+  def get_module_specs(
+      self,
+      data_path: str,
+      additional_flags: Tuple[str, ...] = (),
+      delete_flags: Tuple[str, ...] = ()
+  ) -> List[ModuleSpec]:
+    """Fetch a list of ModuleSpecs for the corpus at data_path
+
+    Args:
+      data_path: base directory of corpus
+      additional_flags: tuple of clang flags to add.
+      delete_flags: tuple of clang flags to remove.
+    """
+    raise NotImplementedError
+
+  @staticmethod
+  def _get_module_specs(data_path: str, additional_flags, delete_flags,
+                        xopts) -> List[ModuleSpec]:
+    module_paths: List[str] = get_module_paths(data_path)
+
+    is_thinlto: bool = is_thinlto_fn(module_paths)
+    has_cmd: bool = has_cmd_fn(module_paths)
+
+    # TODO: (b/233935329) Per-corpus *fdo profile paths can be read into
+    # {additional|delete}_flags here
+    meta = load_metadata(data_path)
+
+    xopts.update({'output': ['-o', '{path:s}']})
+
+    module_specs: List[ModuleSpec] = []
 
-def is_thinlto(module_paths: Iterable[str]) -> bool:
+    # Future: cmd_override could have per-module overrides if needed
+    cmd_override = get_cmd_override(meta, critical=not has_cmd)
+
+    for module_path in module_paths:
```

Comment:
how long does this for loop take? if it's taking too long, may need multithreading/multiprocessing here

Reply:
_xopts looks trivial, but anyway if it's 7s, then not worth doing it.